### ISSUE TITLE
Saving to the same file leads to data loss

### ISSUE THREAD

**Issue by Mrazator** [excalidraw/excalidraw#8841]
**Reproduction**
1. save "old scene"
2. delete all elements (to start from scratch)
3. press ctrl + s

**Actual result**
- Backed up "old scene" got just overridden

**Comment by dwelle** [comment]
This is technically by design. We're not making assumptions about what canvas action constitutes an action that should reset the active file handle — besides explicitly clearing canvas using the main menu.

I'm not sure that deleting all elements on canvas should reset the file handle as that'd be non-standard? E.g., if you delete all characters in a saved text file and save it again, it does the same thing.

**Comment by dwelle** [comment]
But, thinking about it, deleting all elements probably should clear the active file handle to be on the save side, as it's not gonna happen too often to be that annoying, and losing data is worse.

Problem is we'd have to be checking each update whether all elements are deleted and resetting the file handle if so. But we also need to track whether previous update there were some non-deleted elements so that we don't clear the file handle just as you save an empty canvas to start a file (or similar cases).

**Comment by Mrazator** [comment]
> But, thinking about it, deleting all elements probably should clear the active file handle to be on the save side, as it's not gonna happen too often to be that annoying, and losing data is worse.

Yeah, that's the point. Also, clear canvas should do the same (if it does not already, can't check rn).

Still "select all & delete" is a workaround for a missing "create new", which would allow to persist the current scene (to file or E+) or clear it (clear canvas). Either way, it should discard the current file handle.

**Comment by Mrazator** [comment]
> allow to persist the current scene (to file or E+) or clear it (clear canvas)

Essentially combining "save to" and "clear canvas" into one **prominent** button. As discussed in https://github.com/excalidraw/excalidraw/issues/1957, as "save to" + "clear canvas" are two steps and not really intuitive.
